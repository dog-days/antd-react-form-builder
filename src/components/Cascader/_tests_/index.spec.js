import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import Cascader from '../index'

var options = [
  {
    value: 'zhejiang',
    label: 'Zhejiang',
    children: [{
      value: 'hangzhou',
      label: 'Hangzhou',
      children: [{
        value: 'xihu',
        label: 'West Lake',
      }],
    }],
  }, 
  {
    value: 'jiangsu',
    label: 'Jiangsu',
    children: [{
      value: 'nanjing',
      label: 'Nanjing',
      children: [{
        value: 'zhonghuamen',
        label: 'Zhong Hua Men',
      }],
    }],
  }
];

describe('Cascader',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <Cascader 
        required
        className="test"
        label="name"
        name="test"
        options={ options }
        value={
          ["jiangsu","nanjing","zhonghuamen"]
        }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(Cascader); 
  it("should have hidden <Input /> ",function(){
    const wrapper = mount(
      <Cascader 
        required
        className="test"
        label="name"
        name="test"
        value={
          ["jiangsu","nanjing","zhonghuamen"]
        }
      />
    );
    expect(wrapper.childAt(1).childAt(0).props().name).toEqual("test");
    expect(wrapper.childAt(2).childAt(0).props().name).toEqual("test");
    expect(wrapper.childAt(3).childAt(0).props().name).toEqual("test");
  })
})

