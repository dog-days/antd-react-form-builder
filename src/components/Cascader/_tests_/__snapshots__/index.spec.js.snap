// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control has-success"
      >
        <span
          class="test ant-cascader-picker"
          tabindex="0"
        >
          <input
            autocomplete="off"
            class="ant-input ant-cascader-input "
            name="test"
            readonly=""
            type="cascader"
            value=""
          />
          <span
            class="ant-cascader-picker-label"
          >
            Jiangsu / Nanjing / Zhong Hua Men
          </span>
          <i
            class="anticon anticon-cross-circle ant-cascader-picker-clear"
          />
          <i
            class="anticon anticon-down ant-cascader-picker-arrow"
          />
        </span>
      </div>
    </div>
  </div>
  <span>
    <input
      class="ant-input"
      name="test"
      type="hidden"
      value="jiangsu"
    />
  </span>
  <span>
    <input
      class="ant-input"
      name="test"
      type="hidden"
      value="nanjing"
    />
  </span>
  <span>
    <input
      class="ant-input"
      name="test"
      type="hidden"
      value="zhonghuamen"
    />
  </span>
</span>
`;
